addons/project/models/project_update.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import timedelta
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+class ProjectUpdateStatus(models.Model):
+    _name = 'project.update.status'
+    _description = 'Project Update Status'
+    _order = 'default desc, sequence, id'
+
+    def _get_default_project_ids(self):
+        default_project_id = self.env.context.get('default_project_id')
+        return [default_project_id] if default_project_id else None
+
+    name = fields.Char()
+    color = fields.Integer()
+    default = fields.Boolean()
+    sequence = fields.Integer()
+    project_ids = fields.Many2many('project.project', 'project_update_status_rel', 'update_status_id', 'project_id', string='Projects',
+                                   default=_get_default_project_ids)
+
+class ProjectUpdate(models.Model):
+    _name = 'project.update'
+    _description = 'Project Update'
+    _order = 'create_date desc'
+    _inherit = ['mail.thread.cc', 'mail.activity.mixin']
+
+    def _get_default_project_id(self):
+        return self.env.context.get('default_project_id') or self.env.context.get('active_id')
+
+    def _get_default_status(self):
+        project_id = self.env.context.get('default_project_id') or self.env.context.get('active_id')
+        if not project_id:
+            return False
+        return self.env['project.update.status'].search([('project_ids', '=', project_id)], limit=1)
+
+    def _get_default_progress(self):
+        project_id = self.env.context.get('default_project_id') or self.env.context.get('active_id')
+        if not project_id:
+            return False
+        return self.env['project.project'].browse(project_id).last_update_id.progress
+
+    def _get_default_description(self):
+        project_id = self.env.context.get('default_project_id') or self.env.context.get('active_id')
+        if not project_id:
+            return False
+        return self._build_description(project_id)
```

Comment:
Could group this into a override of _default_get to avoid the code repetition and multiple search/browse ? :)

Reply:
Done, thanks ! :)